src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -246,15 +253,49 @@ splitMeshByPartition( vtkUnstructuredGrid & mesh,
 
 vtkSmartPointer< vtkUnstructuredGrid >
 redistributeByCellGraph( vtkUnstructuredGrid & mesh,
+                         VTKMeshGenerator::PartitionMethod const method,
                          MPI_Comm const comm,
                          int const numRefinements )
 {
   GEOSX_MARK_FUNCTION;
 
+  int64_t const numElems = mesh.GetNumberOfCells();
+  int64_t const numProcs = MpiWrapper::commSize( comm );
+
+  // Compute `elemdist` parameter (element range owned by each rank)
+  array1d< int64_t > const elemDist( numProcs + 1 );
+  {
+    array1d< int64_t > elemCounts;
+    MpiWrapper::allGather( numElems, elemCounts, comm );
+    std::partial_sum( elemCounts.begin(), elemCounts.end(), elemDist.begin() + 1 );
+  }
+
   // Use int64_t here to match ParMETIS' idx_t
   ArrayOfArrays< int64_t, int64_t > const elemToNodes = buildElemToNodes< int64_t >( mesh );
-  int64_t const numProcs = MpiWrapper::commSize( comm );
-  array1d< int64_t > const newParts = parmetis::partMeshKway( elemToNodes.toViewConst(), comm, 3, numRefinements );
+  ArrayOfArrays< int64_t, int64_t > const graph = parmetis::meshToDual( elemToNodes.toViewConst(), elemDist, comm, 3 );
+
+  array1d< int64_t > const newParts = [&]()
+  {
+    switch( method )
+    {
+      case VTKMeshGenerator::PartitionMethod::parmetis:
+      {
+        return parmetis::partition( graph.toViewConst(), elemDist, numProcs, comm, numRefinements );
+      }
+      case VTKMeshGenerator::PartitionMethod::ptscotch:
+      {
+#ifdef GEOSX_USE_SCOTCH
+        return ptscotch::partition( graph.toViewConst(), numProcs, comm );
+#else
```

Comment:
Is there an equivalent of `ParMETIS_V3_RefineKway` in PT-Scotch? If not, maybe we can add a warning if someone tries to use a non-zero `numRefinements` with PT-Scotch.

Reply:
No, I didn't find anything comparable in the user manual. My plan was to ignore the refinement level in that case, but I agree a warning is a good idea. In future we might add more options that are specific to either method and I hope to refactor the inputs a bit.